### PR TITLE
[refactor] [net] Clean up InactivityCheck()

### PR DESCRIPTION
**Pull request by jnewbery**
This is a pure refactor and should not change any behavior. It clarifies and documents the InactivityCheck() function

This makes #20721 easier to review. In particular, this function uses a mixture of (unmockable) system time and mockable time. It's important to understand where those are being used when reviewing #20721.

#20721 doesn't require this change, so if others don't agree that it's useful and makes review easier, then I'm happy to close this and just do #20721 directly.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK: I find the suggested version easier to reason about.

Also thanks for switching to `CNode*` to `CNode&`: it makes the currently implicit precondition explicit. As we all know by know: explicit is much better than implicit :)

**Comment by jnewbery** [comment]
I've updated the commit message on the second commit to give a bit more context about why the change to the return type is an improvement.

**Review by mjdietzx — COMMENTED** [review]
crACK 075d3c5ea25e5e4ec9ee1a18e565aeaed16b2214

I like that you made `InactivityChecks` a pure function, and agree this is a little cleaner. nit: I didn't find changing the ordering of the if statements to be clearer, but I don't feel strongly about it

**Comment by jnewbery** [comment]
Thanks for the review @mjdietzx!

> I didn't find changing the ordering of the if statements to be clearer, but I don't feel strongly about it

Maybe it's just me, but I find that standardizing all time comparisons to `now <` or `now >` (or in English - "it is [now] before/after the timeout") makes it easier to intuitively understand the comparisons. I don't feel strongly and I'm happy to change them back if the other reviewers agree with you.

**Comment by jnewbery** [comment]
I've pushed a small update to make the `CNode&` argument const now that the function isn't mutating it.

**Review by fanquake — APPROVED** [review]
ACK bf100f8170770544fb39ae6802175c564cde532f

File: src/net.cpp [hunk 1]
```diff
@@ -1216,37 +1216,47 @@ void CConnman::NotifyNumConnectionsChanged()
...
-            LogPrintf("ping timeout: %fs\n", 0.000001 * count_microseconds(GetTime<std::chrono::microseconds>() - node.m_ping_start.load()));
-            node.fDisconnect = true;
-        }
-        else if (!node.fSuccessfullyConnected)
-        {
-            LogPrint(BCLog::NET, "version handshake timeout from %d\n", node.GetId());
-            node.fDisconnect = true;
-        }
+    // Use non-mockable system time (otherwise these timers will pop when we
+    // use setmocktime in the tests).
```
Comment (MarcoFalke) [inline src/net.cpp:1222]: this can be worked around by mocking m_peer_connect_timeout to a high value

**Review by maflcko — APPROVED** [review]
review ACK bf100f8170770544fb39ae6802175c564cde532f 💫

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

review ACK bf100f8170770544fb39ae6802175c564cde532f 💫
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgT8Av/b4FZwKaQtoksDjtnB2DNCLN+Ldh3haCVJR0xcqjm0RoIPbfWVW+542b3
6EoqZIpl1l1DNyZh2ej3ATRGB9FqQF2McIAZqrboXIpUScMPf/yd0f6rJRi2WoCR
Ga/jHA3gX9jzfjOWQEOdQ/CloK8ic/5WKF9B8BNcR7e+kf+MeFazSyFB4RNFUYqP
8oOt4secmfOChQJgBQMAv0RgS1pN7+MJ/uvIF/USeM+Zw8/VLbxUwmKlL6BUqnM+
sa14lX7nFyI6YC7x45mfdvv3tLDZfsOaYNgpN1K8YYzuuE6dIk3FzTkZJqV/5iUz
isP8xRgh9Upv6GZK1EzTbY57YzJ15WAjYu2qmr3CCxU7J2/ZtYPJPYYsve9CQkT9
ej+75BdjF+Q9d1RgJvIFUUfnuUfVyo1zTnbiTQum/CKot6AuGM1VAAJXvqdIGakN
B5hXgACOPBs0HJ6BBKnwARygNigXMzHiCD5Mfa38UrTtvu5uJklNKTQ1NSOvOOlq
n2I2skbQ
=OdKZ
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `9262a3bf7f8cbef76444cea7a5dd2356354c35fbb3f38dbc8fac626d2cd79705  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e8929401089262a3bf7f8cbef76444cea7a5dd2356354c35fbb3f38dbc8fac626d2cd79705f010d38ffe470ed2986ad5bad9cd3c7f770808fff01043a9b6736844601f108fde0216471a1408f104600ac0f6f008419ff4c39e31d8350083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff0105c9dc3132c768defb1b87cb65f64208708f104600ac0f6f0086f1bc39b01e691d80083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010abb4b79709a5925fac534f7f091d845208f104600ac0f7f00825a1c5d58fd8c1200083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010e3ab519f6dcdfe6a55f8ef55ccae306e08f0205634090ef11dfcdd2946abaf36ccd245bd6359147cf49a9b3c58f7b6d1bcd8a608f104600ac0f7f008ccd10cba073dcc130083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

**Pull request merged**